### PR TITLE
feat(tui): resume saved sessions from the reply composer

### PR DESCRIPTION
**Pull request by snimu**
## What

Pressing space on an **inactive** (saved, non-resident) session now opens the same reply composer as on active sessions. Sending the message resumes the saved conversation into the daemon and delivers the prompt in one motion: the row migrates from Inactive to Running and the reply arrives as the next turn.

## How

The view was already unified at the data layer (`UnifiedSessionRecord`); this unifies the **action layer** where the reply path still branched on `activeSessionId`:

- `resumeSavedAgentsViewSession(client, config, summary)` extracts the create-with-`sessionPath` block from `openAgentsViewSession`, which now composes to "resume, then attach". The daemon's `create` with `sessionPath` is idempotent for the owning client (`reuseWorkerForCreate`), so a session that became active between the catalog refresh and Enter is reused, not double-resumed.
- The reply composer state (`replyActiveSessionId: string`) becomes `replyTarget: { key, summary }` — key stays `activeSessionId ?? id`, matching the selection-follow convention, so the arm/disarm lifecycle (query stash, placeholder, header, Escape/Left) is shared verbatim between live and saved targets.
- `sendReply` resolves the target: live → prompt directly (unchanged); saved → `resumeSavedAgentsViewSession` then the same `sendPrompt`, with selection pointed at the resumed row (identity aliases keep it stable across the id change).
- Preview: live targets keep `get_last_assistant_text`; saved targets use the persisted recap (`summary` / `firstMessage`) with no daemon round-trip.
- Footer hint reads `space resume` on inactive rows; the composer placeholder says "Write a prompt to resume this session".

Local-adapter mode keeps space disabled, as reply already was (nothing to resume into without a daemon).

## Notes

- A cwd-fallback notice from `resolveAgentsViewOpenCwd` is surfaced as a sticky status, same as the open path.
- If the prompt fails after the resume, the session intentionally stays resident (the user asked to wake it); the row shows truthfully in Idle.

## Testing

- New `agents-view-inactive-reply.test.ts`: composer arms on saved-only rows using the recap without a daemon call; rows with neither runtime nor file stay ignored; saved targets resume (create with `sessionPath`) before `prompt` and reselect the resumed row; live targets prompt without resuming.
- `agents-view-state`, `agents-view-missing-cwd`, `interactive-mode-startup` suites pass; `npm run check` and build pass.

---

> [!NOTE]
> **Medium Risk**
> Touches session input pump, prompt admission, and daemon/supervisor prompt paths alongside new resume-on-send UX; behavior changes are covered by new and updated tests but runtime scheduling edge cases warrant review.
> 
> **Overview**
> **Agents view** extends the reply composer to **inactive** saved sessions: arming uses a `replyTarget` (`key` + `summary`) instead of only a live `activeSessionId`, shows a resume placeholder and persisted recap preview without a daemon call, and **send** resumes via shared `resumeSavedAgentsViewSession` then delivers the prompt (with `resolveCurrentReplyTargetSummary` for stale IDs). Submit/selection/status handling covers success, failure, cancellation, and sticky cwd notices; footer hints distinguish **resume** vs **reply**.
> 
> **Session core** moves `waitForPromptAdmission` / `PromptAdmissionCancelledError` into `prompt-admission.ts` (daemon, supervisor, and agent-session import it). `AgentSession` drops legacy `pauseQueuedWork`, makes `_steeringStopPending` a computed getter, tightens deferred session-input handling (busy requeue surfaces errors; extension error emit is try/caught), and simplifies `resumeQueuedWork` to always reschedule the pump.
> 
> **Tests** add `agents-view-inactive-reply.test.ts` and `prompt-admission.test.ts`; queue/restart tests use `waitForSessionInputIdle` / `acquireQueuedWorkPause`; serialized-refine tests seed steering via `_steeringMessages`.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 75e51bc0c151f4f4d13d510d6cebccc3363d35fb. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

> [!NOTE]
> ### Add resume of saved sessions from the reply composer in agents view
> - Users can now arm the reply composer on inactive (saved-only) session rows; submitting the reply transparently resumes the session before delivering the prompt.
> - `AgentsViewMode.sendReply` resolves stale runtime IDs via a new `resolveCurrentReplyTargetSummary` helper, resumes via `resumeSavedAgentsViewSession` (idempotent create-with-sessionPath), then sends the prompt.
> - The composer placeholder changes to "Resume" when the target session is inactive, and help hints clarify resume vs. reply.
> - Prompt admission wait/cancel logic is extracted into a shared [`prompt-admission.ts`](https://github.com/PrimeIntellect-ai/prime-agent/pull/528/files#diff-75f495f4ca7a4305169d34d4b859bdc733947cdd3223557df554595135c50b9c) module, replacing duplicated implementations in the daemon and supervisor.
> - `AgentSession._steeringStopPending` is converted from a mutable flag with side-effect updaters to a computed getter, and the legacy `pauseQueuedWork` API is removed.
>
> 
>
> <sup><a href="https://app.macroscope.com">Macroscope</a> summarized 75e51bc.</sup>
> 
>
>

### CODE REVIEW

File: packages/coding-agent/src/modes/agents-view/agents-view-mode.ts [hunk 1]
```diff
@@ -1338,10 +1373,24 @@ export class AgentsViewMode implements Component, Focusable {
...
+					this.requireClient(),
+					this.options.config,
+					target.summary,
+				);
+				activeSessionId = resumed.activeSessionId;
+				if (resumed.cwdFallbackNotice) this.setStatusMessage(resumed.cwdFallbackNotice, { sticky: true });
+				this.selectSummary(resumed.summary);
+			}
+			const behavior = this.findSummaryByActiveSessionId(activeSessionId)?.isStreaming ? "followUp" : undefined;
+			this.setStatusMessage("Sending reply...");
```
[the remark this answers is not in the document]
Comment (snimu) [reply packages/coding-agent/src/modes/agents-view/agents-view-mode.ts:1436]: Valid. The sticky cwd-fallback notice was set before `"Sending reply..."`, which immediately replaced it, so the warning was never visible on this path.

Fixed by holding the notice and surfacing it as the final status after the prompt is delivered (still sticky), taking the place of the transient `"Reply sent"`. Test added: `keeps the cwd-fallback notice visible after the reply is sent`.

File: packages/coding-agent/src/modes/agents-view/agents-view-mode.ts [hunk 2]
```diff
@@ -1338,10 +1373,24 @@ export class AgentsViewMode implements Component, Focusable {
...
+				const resumed = await resumeSavedAgentsViewSession(
+					this.requireClient(),
+					this.options.config,
+					target.summary,
+				);
+				activeSessionId = resumed.activeSessionId;
+				if (resumed.cwdFallbackNotice) this.setStatusMessage(resumed.cwdFallbackNotice, { sticky: true });
+				this.selectSummary(resumed.summary);
+			}
+			const behavior = this.findSummaryByActiveSessionId(activeSessionId)?.isStreaming ? "followUp" : undefined;
```
[the remark this answers is not in the document]
Comment (snimu) [reply packages/coding-agent/src/modes/agents-view/agents-view-mode.ts]: Valid. After resuming, `findSummaryByActiveSessionId` searched the stale pre-resume rows, returned `undefined`, and the prompt was sent without `followUp` — so a resumed session that immediately re-entered streaming (e.g. restart-queue re-delivery) would reject the reply.

Fixed as suggested: the resume path derives `behavior` from `resumed.summary.isStreaming` (the authoritative post-resume state), while live targets keep the row lookup. The existing resume test now asserts `followUp` is passed when the resume response reports streaming.

File: packages/coding-agent/src/modes/agents-view/agents-view-mode.ts [hunk 3]
```diff
@@ -1338,19 +1373,49 @@ export class AgentsViewMode implements Component, Focusable {
...
+				);
+				activeSessionId = resumed.activeSessionId;
+				// The rows are still pre-resume; the fresh summary is the authoritative
+				// streaming state for scheduling the prompt.
+				liveSummary = resumed.summary;
+				cwdFallbackNotice = resumed.cwdFallbackNotice;
+				this.selectSummary(resumed.summary);
+			}
+			const behavior = liveSummary?.isStreaming ? "followUp" : undefined;
+			this.setStatusMessage("Sending reply...");
```
[the remark this answers is not in the document]
Comment (snimu) [reply packages/coding-agent/src/modes/agents-view/agents-view-mode.ts:1436]: Fixed in `c301a54e4`. Submission still clears the editor while a request is in flight (preventing duplicate Enter sends), but `sendReply()` now reports success/failure and `submit()` owns composer lifecycle. Failure restores the captured draft only when the same target is still armed and the editor is still empty; replacement text is never overwritten. Success likewise closes only the captured target, so an older in-flight send cannot clear a newly armed composer. Tests cover resume failure, prompt failure, replacement text, and target replacement.

File: packages/coding-agent/src/modes/agents-view/agents-view-mode.ts [hunk 4]
```diff
@@ -1338,19 +1373,49 @@ export class AgentsViewMode implements Component, Focusable {
...
 			await this.sendPrompt(activeSessionId, text, behavior);
-			this.setStatusMessage("Reply sent");
+			// The fallback-directory notice must outlive the transient send statuses.
+			if (cwdFallbackNotice) this.setStatusMessage(cwdFallbackNotice, { sticky: true });
+			else this.setStatusMessage("Reply sent");
```
[the remark this answers is not in the document]
Comment (snimu) [reply packages/coding-agent/src/modes/agents-view/agents-view-mode.ts:1440]: Fixed in `c301a54e4`. Immediately after `resumeSavedAgentsViewSession()` succeeds, the captured saved-row identity is removed from `inactiveAgentIdentities` before prompt delivery. This is the correct transition boundary: the runtime is live even if the subsequent prompt fails, while a create/resume failure keeps the manual inactivity fence. The reply regressions cover both sides of that boundary.

File: packages/coding-agent/src/modes/agents-view/agents-view-mode.ts [hunk 5]
```diff
@@ -1338,19 +1380,50 @@ export class AgentsViewMode implements Component, Focusable {
...
+					this.requireClient(),
+					this.options.config,
+					target.summary,
+				);
+				activeSessionId = resumed.activeSessionId;
+				// The rows are still pre-resume; the fresh summary is the authoritative
+				// streaming state for scheduling the prompt.
+				liveSummary = resumed.summary;
+				cwdFallbackNotice = resumed.cwdFallbackNotice;
+				this.inactiveAgentIdentities.delete(getSummaryIdentity(target.summary));
```
[the remark this answers is not in the document]
Comment (snimu) [reply packages/coding-agent/src/modes/agents-view/agents-view-mode.ts:1430]: Fixed in the saved-reply owner. After asynchronous saved-session resume, `sendReply()` selects the resumed summary only if `this.replyTarget === target`; cancelling the composer and moving elsewhere can no longer have its selection stolen by stale completion. Prompt delivery still reaches the resumed live session and successful resume still clears inactivity. The regression covers cancellation and selection movement while resume is pending.

File: packages/coding-agent/src/modes/agents-view/agents-view-mode.ts [hunk 6]
```diff
@@ -1008,11 +1029,18 @@ export class AgentsViewMode implements Component, Focusable {
...
-		if (this.replyActiveSessionId) {
+		if (this.replyTarget) {
+			const target = this.replyTarget;
 			const text = value.trim();
 			if (text) {
 				this.editor.setText("");
-				await this.sendReply(this.replyActiveSessionId, text);
+				const sent = await this.sendReply(target, text);
+				if (sent) {
+					if (this.replyTarget === target) this.setReplyTarget(undefined);
```
[the remark this answers is not in the document]
Comment (snimu) [reply packages/coding-agent/src/modes/agents-view/agents-view-mode.ts]: Fixed in ef943fee9. A successful reply leaves reply mode only when the same target still owns an empty editor. New text typed into that composer while send/resume is in flight is preserved, with focused success-race coverage.

File: packages/coding-agent/src/modes/agents-view/agents-view-mode.ts [hunk 7]
```diff
@@ -1337,19 +1381,55 @@ export class AgentsViewMode implements Component, Focusable {
...
-	private async sendReply(activeSessionId: string, text: string): Promise<void> {
-		const behavior = this.findSummaryByActiveSessionId(activeSessionId)?.isStreaming ? "followUp" : undefined;
-		this.setStatusMessage("Sending reply...");
+	private async sendReply(target: { key: string; summary: SessionSummary }, text: string): Promise<boolean> {
+		let activeSessionId = target.summary.activeSessionId;
+		let liveSummary = this.findSummaryByActiveSessionId(activeSessionId ?? target.key);
+		let cwdFallbackNotice: string | undefined;
+		let didResume = false;
 		try {
+			if (!activeSessionId) {
```
[the remark this answers is not in the document]
Comment (snimu) [reply packages/coding-agent/src/modes/agents-view/agents-view-mode.ts:1415]: Fixed in 01d808a61. Before sending, the composer target’s stable persisted identity is resolved against the latest unified catalog. Delivery uses the current live active ID when present; if the current row is saved-only (or the captured live runtime disappeared), it resumes the current persisted summary instead of prompting the stale active ID. Added a live-to-inactive composer regression.

File: packages/coding-agent/src/core/agent-session.ts [hunk 8]
```diff
@@ -5145,7 +5101,14 @@ export class AgentSession {
 	}
 
 	private _isDeferredSessionInputError(error: unknown, epoch: number): boolean {
```
[the remark this answers is not in the document]
Comment (snimu) [reply packages/coding-agent/src/core/agent-session.ts:5143]: Fixed in c908425c7: _surfaceSessionInputError now guards the emitError call, so a throwing error listener cannot escape before the pump requeues undelivered prompts.

**Comment by snimu** [comment]
Consolidated into #541 for review. This PR is kept closed (not deleted) as the focused layer description and review history for its slice of the stack; the branch stays intact.
